internal/gateway/statistics/statistics.go

```diff
@@ -0,0 +1,69 @@
+package statistics
+
+import (
+	"fmt"
+	"os/exec"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+type Counters struct {
+	ChunksAdded          int64  `json:"Publish.n_chunks_added,string"`
+	ChunksDuplicated     int64  `json:"Publish.n_chunks_duplicated,string"`
+	CatalogsAdded        int64  `json:"Publish.n_catalogs_added,string"`
+	UploadedBytes        int64  `json:"Publish.sz_uploaded_bytes,string"`
+	UploadedCatalogBytes int64  `json:"Publish.sz_uploaded_catalog_bytes,string"`
+	StartTime            string `json:"start_time"`
+}
+
+// TODO: concurrent access
+type StatisticsMgr struct {
+	leaseCounters map[string]Counters
+}
+
+func NewStatisticsMgr() (*StatisticsMgr, error) {
+	return &StatisticsMgr{leaseCounters: make(map[string]Counters)}, nil
+}
+
+func (m *StatisticsMgr) AddLease(leasePath string, counters Counters) error {
+	m.leaseCounters[leasePath] = counters
```

Comment:
This is never used, maybe it make sense to have it private? It is more than fine also like this, it is just a nitpick... Feel free to ignore it.